pkg/sourcer/service.go

```diff
@@ -140,8 +141,9 @@ func (fs *Service) receiveReadRequests(ctx context.Context, stream sourcepb.Sour
 			close(messageCh)
 		}()
 		request := readRequest{
-			count:   req.Request.GetNumRecords(),
-			timeout: time.Duration(req.Request.GetTimeoutInMs()) * time.Millisecond,
+			count:    req.Request.GetNumRecords(),
+			timeout:  time.Duration(req.Request.GetTimeoutInMs()) * time.Millisecond,
+			metadata: fromProto(req.Request.GetMetadata()),
```

Comment:
Metadata should be originated from source, request doesn't need to expose metadata we can inject the system metadata after the messages are read.